tests/ops/qubit/test_matrix_ops.py

```diff
@@ -576,6 +578,244 @@ def test_matrix_already_in_cache_list(self, mat, op):
+        assert op.hyperparameters["norm"] == expected_hyperparameters["norm"]
+        assert op.hyperparameters["subspace"] == expected_hyperparameters["subspace"]
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [(1, 1), (1, 2), (1, [1]), (1, range(2)), (np.identity(2), ["a", "b"])],
+    )
+    def test_varied_wires(self, input_matrix, wires):
+        """Test that BlockEncode wires are stored correctly for various wire input types."""
+        assert qml.BlockEncode(input_matrix, wires).wires == Wires(wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "msg"),
+        [
+            (
+                [[0, 1], [1, 0]],
+                1,
+                r"Block encoding a \(2 x 2\) matrix requires a hilbert space of size"
+                r" at least \(4 x 4\). Cannot be embedded in a 1 qubit system.",
+            ),
+        ],
+    )
+    def test_error_raised_invalid_hilbert_space(self, input_matrix, wires, msg):
+        """Test the correct error is raised when inputting an invalid number of wires."""
+        with pytest.raises(ValueError, match=msg):
+            qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "output_matrix"),
+        [
+            (1, 0, [[1, 0], [0, -1]]),
+            (0.3, 0, [[0.3, 0.9539392], [0.9539392, -0.3]]),
+            (
+                [[0.1, 0.2], [0.3, 0.4]],
+                range(2),
+                [
+                    [0.1, 0.2, 0.97283788, -0.05988708],
+                    [0.3, 0.4, -0.05988708, 0.86395228],
+                    [0.94561648, -0.07621992, -0.1, -0.3],
+                    [-0.07621992, 0.89117368, -0.2, -0.4],
+                ],
+            ),
+            (
+                0.1,
+                range(2),
+                [[0.1, 0.99498744, 0, 0], [0.99498744, -0.1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]],
+            ),
+        ],
+    )
+    def test_correct_output_matrix(self, input_matrix, wires, output_matrix):
+        """Test that BlockEncode outputs the correct matrix."""
+        assert np.allclose(qml.matrix(qml.BlockEncode)(input_matrix, wires), output_matrix)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [
+            (1, 0),
+            (0.3, 0),
+            (np.array([[0.1, 0.2], [0.3, 0.4]]), range(2)),
+            (np.array([[0.1, 0.2, 0.3]]), range(2)),
+            (np.array([[0.1], [0.2], [0.3]]), range(2)),
+            (np.array([[0.1, 0.2], [0.3, 0.4], [0.5, 0.6]]), range(3)),
+            (np.array([[1, 2], [3, 4]]), range(2)),
+        ],
+    )
+    def test_unitary(self, input_matrix, wires):
+        """Test that BlockEncode matrices are unitary."""
+        mat = qml.matrix(qml.BlockEncode(input_matrix, wires))
+        assert np.allclose(np.eye(len(mat)), mat.dot(mat.T.conj()))
+
+    # TODO: add tensorflow support
+    # def test_blockencode_tf(self):
```

Comment:
You may leave the #TODO, but I suggest removing the other lines of commented code.